include/picongpu/fields/incidentField/profiles/GaussianPulse.hpp

```diff
@@ -34,328 +34,345 @@
 #include <type_traits>
 
 
-namespace picongpu
+namespace picongpu::fields::incidentField
 {
-    namespace fields
+    namespace profiles
     {
-        namespace incidentField
+        namespace detail
         {
-            namespace profiles
+            /** Base class providing tilt value based on given parameters
+             *
+             * General implementation sets tilts to 0 and does not require T_Params having tilt as member.
+             *
+             * @tparam T_Params user (SI) parameters
+             */
+            template<typename T_Params, typename T_Sfinae = void>
+            struct TiltParam
             {
-                namespace detail
+                // unit: radian
+                static constexpr float_X TILT_AXIS_1 = 0.0_X;
+                // unit: radian
+                static constexpr float_X TILT_AXIS_2 = 0.0_X;
+            };
+
+            /** Helper type to check if T_Params has members TILT_AXIS_1_SI and TILT_AXIS_2_SI.
+             *
+             * Is void for those types, ill-formed otherwise.
+             *
+             * @tparam T_Params user (SI) parameters
+             */
+            template<typename T_Params>
+            using HasTilt = std::void_t<decltype(T_Params::TILT_AXIS_1_SI + T_Params::TILT_AXIS_2_SI)>;

```

Comment:
What does this do and what happens if `TILT_AXIS_1_SI` ≠ 0 and `TILT_AXIS_2_SI` ≠ 0 but the sum =0? 

Above you define the variable without `_SI` here you use it with `_SI`. Where do you define the later one?

Reply:
Code that existed before